### DISCUSSION TITLE
Any known problems running under Wine on Windows?

### DISCUSSION THREAD

**Discussion by John-Nagle** [rust-windowing/winit#3403] [Q&A]
I recently updated some of my programs to a later version of Rend3, which pulled in a new version of winit (0.29.8). For some reason, winit keyboard events are delayed about 2-4 seconds when running on a cross-compiled executable built on Linux and compiled for a Windows target, then run under Wine. Works OK on Linux. Works OK on actual Windows 10. But something doesnt play well with the event queue used by Wine.

This worked in older versions of winit, but I'll have to go back and check to find a good version. Anyone else seeing this?

**Comment by notgull** [comment]
Huh, kinda neat to get an honest to god celebrity in these parts!

`winit` isn't actively tested on Wine as far as I know, and version 0.29.0 introduced new changes in how the event loop and keyboard subsystems worked. So I think I would start there.

**Reply by John-Nagle** [reply]
Yes. Graphics programs which refresh at top speed and never go idle may be stalling out Wine's version of the windows event loop, or something like that. More later.
